compiler/semexprs.nim

```diff
@@ -1422,7 +1422,10 @@ proc dotTransformation(c: PContext, n: PNode): PNode =
   else:
     var i = considerQuotedIdent(c, n[1], n)
     result = newNodeI(nkDotCall, n.info)
-    result.flags.incl nfDotField
+    # we only want to do the dotField transform once per node, so the new node
+    # should not keep the nfDotField flag, as it's the result of a transform.
+    if nfDotField notin n.flags:
+      result.flags.incl nfDotField
```

Comment:
```suggestion
    if nfDotField notin n.flags:
      result.flags.incl nfDotField
    else:
      result.flags.excl nfDotField
```

Is this what was meant here? Because the current behavior is no different from the previous behavior

Reply:
No, I was tired and it should not include the flag as it was in the node before. The transform has already happened and it should be "consumed".